mlrun/datastore/dbfs_store.py

```diff
@@ -0,0 +1,159 @@
+
+        self._filesystem = fsspec.filesystem("dbfs", **self.get_storage_options())
+        return self._filesystem
+
+    def get_storage_options(self):
+        return dict(
+            token=self._get_secret_or_env("DATABRICKS_TOKEN"), instance=self.endpoint
+        )
+
+    def path_and_system_validator(self, key: str):
+        if not self._filesystem:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Performing actions on data-item without a valid filesystem"
+            )
+        if not key.startswith("/"):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Invalid path(key attribute) - path must start with '/'"
+            )
+
+    def get(self, key: str, size=None, offset=0) -> bytes:
+        self.path_and_system_validator(key)
+        size_less_then_one_exception = "negative or zero size argument 1 is invalid."
+        if offset:
+            if size is None:
+                size = 1000000  # The maximum number of allowed bytes to read is 1MB.
+            elif size <= 0:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    size_less_then_one_exception
+                )
+            end = offset + size
+            return self._filesystem._get_data(key, start=offset, end=end)
+        elif size is not None:
+            if size <= 0:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    size_less_then_one_exception
+                )
+            else:
+                end = offset + size
+                return self._filesystem._get_data(key, start=offset, end=end)
+        else:
+            return self._filesystem.cat_file(key)
+
+    def put(self, key, data, append=False):
+
+        self.path_and_system_validator(key)
+        if append:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Append mode not supported for Databricks file system!"
+            )
+        #  can not use append mode because it overrides data.
+        mode = "w"
+        if isinstance(data, bytes):
+            mode += "b"
+        elif not isinstance(data, str):
+            raise TypeError(
+                "Data type unknown.  Unable to put in Databricks file system!"
+            )
+        with self._filesystem.open(key, mode) as f:
+            f.write(data)
+
+    def upload(self, key: str, src_path: str):
+        self.path_and_system_validator(key)
+        self._filesystem.put_file(src_path, key, overwrite=True)
+
+    def stat(self, key: str):
+        self.path_and_system_validator(key)
+        files = self._filesystem.ls(key, detail=True)
+        if len(files) == 1 and files[0]["type"] == "file":
+            size = files[0]["size"]
+        elif len(files) == 1 and files[0]["type"] == "directory":
+            raise FileNotFoundError("Operation expects a file not a directory!")
+        else:
+            raise ValueError("Operation expects to receive a single file!")
```

Comment:
```suggestion
            raise ValueError(f"Operation expects to receive a single file, instead received {len(files)}")
```

Reply:
This is the standard error message. Look at azure_blob/ google_cloud_storage.

Reply:
standard you mean other fs does it? well I'd say worth changing them all to be a bit more indicative, but not certainly on this PR